### PR TITLE
pawn: init at 4.0.5749

### PR DESCRIPTION
**Pull request by panurg**
###### Motivation for this change
Just another scripting language and its compiler, VM libraries and some useful tools.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/pawn/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      machine.
+    '';
+    homepage = https://www.compuphase.com/pawn/pawn.htm;
+    changelog = https://www.compuphase.com/pawn/pawnhistory.htm;
+    license = [
+      licenses.asl20
+      {
+        fullName = "An exception clause to explicitly permit static linking of the library to commercial application";
+        url = "https://www.compuphase.com/pawn/pawn.htm#APACHE_EXCEPTION";
+      }
```
Comment (roberth) [inline pkgs/development/compilers/pawn/default.nix:42]: @alyssais, how should this be represented?

File: pkgs/development/compilers/pawn/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+stdenv.mkDerivation rec {
+  pname = "pawn";
+  version = "4.0.5749";
+
+  src = fetchzip {
+    url = "https://www.compuphase.com/${pname}/${pname}-${version}.zip";
+    sha256 = "1qj2n0xql2bmgvng7q7xf7113bgi2kqgaq9jp0dmys3nc3vm123f";
+    stripRoot = false;
+  };
+  buildInputs = with pkgs; [ cmake gcc_multi ];
```
Comment (AndersonTorres) [inline pkgs/development/compilers/pawn/default.nix:12]: Why not to declare them on function arguments?

Also, `cmake` usually goes in `nativeBuildInputs`.

File: pkgs/development/compilers/pawn/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  meta = with stdenv.lib; {
+    description = "A simple, typeless, 32-bit “scripting” language with a C-like syntax";
+    longDescription = ''
+      Pawn is a simple, typeless, 32-bit extension language with a C-like syntax.
+      The Pawn compiler outputs P-code (or bytecode) that subsequently runs on an
+      abstract machine. Execution speed, stability, simplicity and a small footprint
+      were essential design criterions for both the language and the abstract
+      machine.
+    '';
+    homepage = https://www.compuphase.com/pawn/pawn.htm;
```
Comment (AndersonTorres) [inline pkgs/development/compilers/pawn/default.nix:35]: Put double quotes in the HTTP link strings.

File: pkgs/development/compilers/pawn/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "pawn";
+  version = "4.0.5749";
+
+  src = fetchzip {
+    url = "https://www.compuphase.com/${pname}/${pname}-${version}.zip";
+    sha256 = "1qj2n0xql2bmgvng7q7xf7113bgi2kqgaq9jp0dmys3nc3vm123f";
+    stripRoot = false;
+  };
```
Comment (AndersonTorres) [inline pkgs/development/compilers/pawn/default.nix:11]: A cosmetic whitespace here, please.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -6539,6 +6539,12 @@
     githubId = 1788628;
     name = "pandaman";
   };
+  panurg = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:6542]: Please put this into a separate commit with the message ``maintainers: add panurg``.

File: pkgs/development/compilers/pawn/default.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchzip, pkgs }:
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/pawn/default.nix:1]: Never input pkgs.

File: pkgs/development/compilers/pawn/default.nix [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    cp pawncc $out/bin
+    cp stategraph $out/bin
+    cp pawndisasm $out/bin
+    cp pawndbg $out/bin
+    cp pawnrun $out/bin
+    mkdir -p $out/lib
+    cp *.so $out/lib
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/pawn/default.nix:26]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
I have tried it but it does not work.

**Pull request closed without merging**